keyboards/model01/leds.c

```diff
@@ -16,42 +16,153 @@
 #include <quantum.h>
 #include <i2c_master.h>
 #include <led_tables.h>
+#include <rgb_matrix.h>
+#include <string.h>
 #include "model01.h"
 
 #define I2C_TIMEOUT     1000
 
-#define LINCOR(i) pgm_read_byte(&CIE1931_CURVE[i])
-
-int set_all_leds_to_raw(uint8_t r, uint8_t g, uint8_t b) {
+void set_all_leds_to(uint8_t r, uint8_t g, uint8_t b) {
   uint8_t buf[] = {
     TWI_CMD_LED_SET_ALL_TO,
     b, g, r
   };
-  int ret = 0;
-  ret |= i2c_transmit(I2C_ADDR(LEFT), buf, sizeof(buf), I2C_TIMEOUT);
-  ret |= i2c_transmit(I2C_ADDR(RIGHT), buf, sizeof(buf), I2C_TIMEOUT);
+  i2c_transmit(I2C_ADDR(LEFT), buf, sizeof(buf), I2C_TIMEOUT);
+  i2c_transmit(I2C_ADDR(RIGHT), buf, sizeof(buf), I2C_TIMEOUT);
   _delay_us(10);
-  return ret;
-}
-
-int set_all_leds_to(uint8_t r, uint8_t g, uint8_t b) {
-  return set_all_leds_to_raw(LINCOR(r), LINCOR(g), LINCOR(b));
 }
 
-int set_led_to_raw(uint8_t led, uint8_t r, uint8_t g, uint8_t b) {
+void set_led_to(int led, uint8_t r, uint8_t g, uint8_t b) {
   uint8_t buf[] = {
     TWI_CMD_LED_SET_ONE_TO,
     led & 0x1f,
     b, g, r
   };
   int hand = (led >= 32) ? RIGHT : LEFT;
-  int ret = i2c_transmit(I2C_ADDR(hand), buf, sizeof(buf), I2C_TIMEOUT);
+  i2c_transmit(I2C_ADDR(hand), buf, sizeof(buf), I2C_TIMEOUT);
   _delay_us(10);
-  return ret;
 }
 
-int set_led_to(uint8_t led, uint8_t r, uint8_t g, uint8_t b) {
-  return set_led_to_raw(led, LINCOR(r), LINCOR(g), LINCOR(b));
+#ifdef RGB_MATRIX_ENABLE
+
+const rgb_led g_rgb_leds[DRIVER_LED_TOTAL] = {
```

Comment:
Also, only 10 "modifier" keys? and I'm guessing that they're the thumb keys.

Shouldn't the outer and inner columns be modifiers too?

Reply:
In the default keymap all the actual modifiers are on the thumb keys. Of course space and backspace are not technically modifiers.

When the code says "modifiers" does it actually mean "non-alphanumeric keys"?

Reply:
Well, I think this is more of a personal choice thing, so ... it really depends on what you want. 

The modifier boolean does affect how some of the rgb modes work.  (or like with me, I use the "mod" LEDs for layer indication).   

So, I guess that's your call. 

If you want, you could set it as weak, and that would allow users to re-set it.

Reply:
Either way, let me know what you want to do here.